src/Val.h

```diff
@@ -1711,7 +1711,6 @@ UNDERLYING_ACCESSOR_DEF(TypeVal, zeek::Type*, AsType)
 extern ValPtr check_and_promote(ValPtr v, const TypePtr& new_type, bool is_init,
                                 const detail::Location* expr_location = nullptr);
 
-extern bool same_val(const Val* v1, const Val* v2);
```

Comment:
This is in the public namespace so it needs to have a deprecation cycle.

Reply:
Check out its implementation and see if you still think so!